Hw#9(unit tests)

StasonLV opened the pull request.

StasonLV committed:
тесты добавлены

StasonLV committed:
cherry pick from #8

StasonLV committed:
lint fix

StasonLV committed:
добавил алерты, тесты для алертов

StasonLV committed:
Добавил тесты респонзов в интерактор

StasonLV committed:
добавлено сравнение модели в презентере

bsciosedu commented on TraineeAppTests/NoteListSceneTests/NoteListInteractorTests.swift:
```diff
@@ -0,0 +1,106 @@
+                self.presenter.isCalledPresentNotes,
+                "Получили город. Ждем флаг true"
+            )
+            expectation.fulfill()
```
тут нужно еще проверить то что success правильный ушел в презентер

bsciosedu commented on TraineeAppTests/NoteListSceneTests/NoteListWorkerTests.swift:
```diff
@@ -0,0 +1,42 @@
+
+    func testsFetchWasCalled() {
+        let expectation = expectation(description: "async test fetch models")
+        interactor.fetchResponse = {
```
не не, должны в тестах воркера вызываться методы воркера

StasonLV committed:
Добавил проверку в интерактор тестах, добавил обработку всех кейсов в воркер тестах

StasonLV committed:
naming

StasonLV committed:
minor

StasonLV committed:
Merge branch 'main' into HW#9(UnitTests)
